### PR TITLE
build: Add 'show-arches' target

### LINKED ISSUE #1009: Add make target to list available architectures
Add a `Makefile` target to allow the list of available architectures to be displayed.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a new build target that displays all available
architectures and indicates the default:

```
make show-arches
```

Fixes #1009.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
This should dtrt when a fix for #1008 lands.

File: Makefile [hunk 1]
```diff
@@ -535,9 +543,10 @@ show-usage: show-header
 	@printf "\tdefault             : same as just \"make\".\n"
 	@printf "\tgenerate-config     : create configuration file.\n"
 	@printf "\tinstall             : install files [2].\n"
-	@printf "\tinstall-$(CC_TYPE)-system   : install using standard $(CC_PROJECT_NAME) system paths\n"
-	@printf "\tinstall-$(KATA_TYPE)-system : install using standard $(KATA_PROJECT_NAME) system paths\n"
-	@printf "\tshow-summary        : show install locations\n"
+	@printf "\tinstall-$(CC_TYPE)-system   : install using standard $(CC_PROJECT_NAME) system paths.\n"
+	@printf "\tinstall-$(KATA_TYPE)-system : install using standard $(KATA_PROJECT_NAME) system paths.\n"
```
Comment (grahamwhaley) [inline Makefile]: sneaky '.'s threw me for a moment ;-)

Comment (jodh-intel) [reply Makefile]: They *are* on a different commit though ;)

**Comment by jodh-intel** [comment]
Hi @devimc - could you take a look please?

**Comment by devimc** [comment]
lgtm

**Pull request merged**